ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -204,3 +204,145 @@ def _configure_bgp_mlag_peer_group(self) -> bool:
     @cached_property
     def _evpn_multicast(self) -> bool:
         return get(self._hostvars, "switch.evpn_multicast") is True
+
+    @cached_property
+    def _wan_vrfs(self) -> list:
+        """
+        Return a list of WAN VRFs based on filtered tenants and the AVT.
+
+        Always render VRF on AutoVPN RRs / CV Pathfinder pathfinders
+        """
+        wan_vrfs = []
+
+        # TODO replace this with VRFs fact once it has been implemented
+        network_services_vrfs = {vrf["name"] for tenant in self._filtered_tenants for vrf in tenant["vrfs"]}
+
+        for avt_vrf in get(self._hostvars, "wan_virtual_topologies.vrfs", []):
+            vrf_name = avt_vrf["name"]
+            if vrf_name in network_services_vrfs or self.shared_utils.wan_role == "server":
+                # Needed because we can be rendering either for AutoVPN or CV Pathfinder and key names are different.
+                policy_key = "policy" if self.shared_utils.cv_pathfinder_role else "path_selection_policy"
+                policy_name = get(avt_vrf, "policy", required=True)
+
+                wan_vrf = {
+                    "name": vrf_name,
+                    policy_key: policy_name,
+                }
+
+                # For CV Pathfinder, it is required to go through all the AVT profiles in the policy to assign an ID.
+                if self.shared_utils.cv_pathfinder_role:
+                    # Need to allocate an ID for each profile in the policy, for now picked up from the input.
+                    policy = get_item(
+                        self._cv_pathfinder_policies,
```

Comment:
Move the called function to utils.py

Reply:
moved the calling function to this file instead